scalatest/src/main/scala/org/scalatest/InsertionOrderSet.scala

```diff
@@ -17,7 +17,7 @@ package org.scalatest
 
 private[scalatest] object InsertionOrderSet {
   def apply[A](elements: List[A]): Set[A] =
-    scala.collection.immutable.SortedSet(elements: _*)(new Ordering[A] {
-      def compare(x: A, y: A): Int = elements.indexOf(x) compare elements.indexOf(y)
-    })
+    scala.collection.immutable.SortedSet(elements: _*)(new Ordering[A | Null] {
```

Comment:
**Java template class requires nullable type**
In order for operator overrides to be valid, java templates classes must use `T | Null` for the type. The nullness is then cast away using `asInstanceOf[ClassName[T]]`.